types/experiments.ts

```diff
@@ -0,0 +1,72 @@
+/** All fields are publically accessible via the API and available to clients */
+export interface Experiment {
+  /** Unique identifier for the experiment */
+  slug: string;
+  /** Publically-accesible name of the experiment */
+  public_name: string;
+  /** Short public description of the experiment */
+  public_description: string;
+  /** Experimenter URL */
+  experiment_url: string;
+
+  /** Is the experiment currently live in production? i.e., published to remote settings? */
+  is_published: boolean;
+  /** Are we continuing to enroll new users into the experiment? */
+  is_enrollment_paused: boolean;
+
+  /** Reference to an Audience, which contains targeting information */
+  audience: Audience;
+  /** Bucketing configuration */
+  bucket_config: BucketConfig;
+
+  /** A list of cusom feature metrics relevant to the experiment analysis */
+  feature_metrics: Array<FeatureMetric>;
+  /** Branch configuration for the experiment */
+  branches: Array<Branch>;
+
+  /** Actual publish date of the experiment */
+  start_date: Date;
+  /** Actual end date of enrollment */
+  enrollment_end_date: Date;
+  /** Actual end date of the experiment */
+  end_date: Date;
```

Comment:
Is there a way to specify in this spec that these fields will be optionally `null`?

Reply:
Just `| null` for nullable and `?` for optional I think – should we go with nullable here?

Reply:
What would be the use case for a null `end_date`?

Reply:
I'd suggest nullable over optional. It is generally more convenient in Typescript when the property definitely exists vs if it may or may not be there.

Reply:
@pdehaan It depends on what we want the fields to represent.  Proposed dates we can know when the experiment launchces, actual dates we don't know until we hit them.  So if this is actual end_date, it won't have a value until it actually ends.  In some places in Experimenter we fold those two things together and just call it 'end date' and behind the scenes it's (actual end date, proposed end date, None) in that order of precedence.  So I'm not sure which we want to do here, but regardless these fields should be `null`able in any case of them isn't set at some point.